src/web/app/pages-student/student-profile-page/student-profile-page.component.ts

```diff
@@ -112,25 +115,41 @@ export class StudentProfilePageComponent implements OnInit {
    * Opens a modal box to upload/edit profile picture.
    */
   onUploadEdit(): void {
-    const modalRef: NgbModalRef = this.ngbModal.open(UploadEditProfilePictureModalComponent);
-    modalRef.componentInstance.profilePicLink = this.profilePicLink;
-    modalRef.result.then((formData: FormData) => {
-      if (!formData) {
-        this.statusMessageService.showWarningMessage('No photo uploaded');
-        return;
-      }
-
-      this.httpRequestService.post('/student/profilePic', {}, formData)
-          .subscribe(() => {
-            this.statusMessageService.showSuccessMessage('Your profile picture has been saved successfully');
-
-            // force reload
-            const timestamp: number = (new Date()).getTime();
-            this.profilePicLink = `${this.backendUrl}/webapi/student/profilePic?${timestamp}&user=${this.id}`;
-          }, (response: ErrorMessageOutput) => {
-            this.statusMessageService.showErrorMessage(response.error.message);
-          });
-    }, () => {});
+    this.getProfilePicture()
+        // Return null if no picture is found
+        .pipe(
```

Comment:
Any reason why we don't combine pipe together into one block?
```
.pipe(
    catchError()
    flatMap()
    catchError()
    xxxx
)
```

Reply:
I previously thought that catchError was a 'catch all' function for pipe, but looks like it is not the case. I will make the changes, thanks for pointing it out!